File: send.php
<?php
$curl = curl_init();
$app_id = "x";
$app_sec = "y";
$app_hash  = base64_encode("$app_id:$app_sec");
$messages = [];
$messages["messages"] = [];
$messages["messages"][0]["text"] = "test php";
$messages["messages"][0]["numbers"][] = "966xxxx";
$messages["messages"][0]["sender"] = "test";

curl_setopt_array($curl, array(
    CURLOPT_URL => 'https://api-sms.4jawaly.com/api/v1/account/area/sms/send',
    CURLOPT_RETURNTRANSFER => true,
    CURLOPT_ENCODING => '',
    CURLOPT_MAXREDIRS => 10,
    CURLOPT_TIMEOUT => 0,
    CURLOPT_FOLLOWLOCATION => true,
    CURLOPT_HTTP_VERSION => CURL_HTTP_VERSION_1_1,
    CURLOPT_CUSTOMREQUEST => 'POST',
    CURLOPT_POSTFIELDS =>json_encode($messages),
    CURLOPT_HTTPHEADER => array(
        'Accept: application/json',
        'Content-Type: application/json',
        'Authorization: Basic '.$app_hash
    ),
));

$response = curl_exec($curl);
curl_close($curl);
var_dump(json_decode($response));

File: C#/untitled folder/send.cs
using System;
using System.Net;
using System.IO;
using Newtonsoft.Json.Linq;

class Program {
    static void Main(string[] args) {
        string appId = "api key";
        string appSec = "api secret";
        string appHash = Convert.ToBase64String(System.Text.Encoding.UTF8.GetBytes($"{appId}:{appSec}"));
        
        var request = (HttpWebRequest)WebRequest.Create("https://api-sms.4jawaly.com/api/v1/account/area/sms/send");
        request.Method = "POST";
        request.Headers["Authorization"] = "Basic " + appHash;
        request.ContentType = "application/json";

        using (var streamWriter = new StreamWriter(request.GetRequestStream())) {
            string json = "{\"messages\":[{\"text\":\"test\",\"numbers\":[\"96650000\"],\"sender\":\"test\"}]}";
            streamWriter.Write(json);
        }

        string responseContent;
        try {
            using (WebResponse response = request.GetResponse()) {
                using (Stream responseStream = response.GetResponseStream()) {
                    using (StreamReader streamReader = new StreamReader(responseStream)) {
                        responseContent = streamReader.ReadToEnd();
                    }
                }
            }
        } catch (WebException e) {
            using (WebResponse response = e.Response) {
                using (Stream responseStream = response.GetResponseStream()) {
                    using (StreamReader streamReader = new StreamReader(responseStream)) {
                        responseContent = streamReader.ReadToEnd();
                    }
                }
            }
        }

        JObject responseJson = JObject.Parse(responseContent);

        if (responseJson["code"].ToString() == "200") {
            if (responseJson["messages"][0].Type == JTokenType.Object && responseJson["messages"][0]["err_text"] != null) {
                Console.WriteLine(responseJson["messages"][0]["err_text"].ToString());
            } else {
                Console.WriteLine("تم الإرسال بنجاح");
                Console.WriteLine("job_id: " + responseJson["job_id"].ToString());
            }
        } else if (responseJson["code"].ToString() == "400") {
            Console.WriteLine("error 400");
            Console.WriteLine(responseJson["message"].ToString());
        } else {
            Console.WriteLine("error " + responseJson["code"].ToString());
        }
    }
}
